### PR TITLE
OpenCollective 2.0

### PR DESCRIPTION
**Pull request by maticzav**
This PR introduces all changes discussed in #5, #8.

### CODE REVIEW

File: package.json [hunk 1]
```diff
@@ -9,23 +9,29 @@
     "coverage": "yarn codecov"
   },
   "dependencies": {
+    "@types/lodash.uniq": "^4.5.5",
+    "isomorphic-fetch": "^2.2.1",
```
Comment (znarf) [inline package.json]: No need for isomorphic-fetch if we're just on the backend side?

Comment (maticzav) [reply package.json]: Switched to `node-fetch`, thanks for pointing it out!

File: tests/validate.test.ts [hunk 2]
```diff
@@ -0,0 +1,91 @@
+import { safeDump } from 'js-yaml'
+import { AddressInfo, Server } from 'net'
+import request from 'request-promise-native'
```
Comment (znarf) [inline tests/validate.test.ts:5]: Why not use fetch for everything?

Comment (maticzav) [reply tests/validate.test.ts:5]: It's only part of the dev environment, I am just used to using it for tests.

**Pull request merged**